framework/src/org/checkerframework/framework/util/AnnotationBuilder.java

```diff
@@ -147,6 +148,25 @@ public String toString() {
         };
     }
 
+    /**
+     * Copies every element value from the given annotation to the annotation to be built if the
+     * element name match
+     */
+    public AnnotationBuilder copyElementValuesFromAnnotation(AnnotationMirror annotation) {
+        for (Map.Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
+                annotation.getElementValues().entrySet()) {
+            Name elemName = entry.getKey().getSimpleName();
+            for (ExecutableElement elt :
+                    ElementFilter.methodsIn(annotationElt.getEnclosedElements())) {
+                if (elt.getSimpleName().contentEquals(elemName)) {
+                    elementValues.put(elt, entry.getValue());
+                    break;
+                }
+            }
```

Comment:
I agree with Martin that some error handling would be better than silently ignoring attributes.
It should be enough to keep track whether for each outer loop element you found an inner loop element; if not, raise an error. Some javadoc above would also need updating. The error message should say what to do in that case - i.e. write your own addAliasedAnnotation implementation.